dataflow/snippets/src/main/java/com/example/dataflow/PubSubWriteWithAttributes.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.dataflow;
+
+// [START dataflow_pubsub_write_with_attributes]
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubIO;
+import org.apache.beam.sdk.io.gcp.pubsub.PubsubMessage;
+import org.apache.beam.sdk.options.Description;
+import org.apache.beam.sdk.options.PipelineOptions;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.transforms.MapElements;
+import org.apache.beam.sdk.values.TypeDescriptor;
+
+public class PubSubWriteWithAttributes {
+  public interface Options extends PipelineOptions {
+    @Description("The Pub/Sub topic to write to. Format: projects/<PROJECT>/topics/<TOPIC>")
+    String getTopic();
+
+    void setTopic(String value);
+  }
+
+  // Write messages to a Pub/Sub topic.
+  public static void main(String[] args) {
+    final List<String> messages = Arrays.asList("message1", "message2", "message3", "message4");
+
+    var options = PipelineOptionsFactory.fromArgs(args).withValidation().as(Options.class);
```

Comment:
Please modify the sample to follow the [minimal arguments](https://googlecloudplatform.github.io/samples-style-guide/#minimal-arguments) guidelines. In its current form, the sample requires additional explanations to figure out the parameters. I recommend to host the code samples in a separate function with explicit parameters and to use `main()` as a wrapper to call this function if required (although it is not necessary). Arguments to the public method make the parameters more verbose and easy to understanding from the code.

Reply:
That's idiomatic for Dataflow, to pass in arbitrary arguments to `PipelineOptionsFactory.fromArgs`, so the caller can set [pipeline options](https://beam.apache.org/documentation/programming-guide/#pipeline-options-cli), which can vary depending on the Beam runner being used.

(See examples of this in the [Beam docs](https://beam.apache.org/documentation/programming-guide/#creating-a-pcollection).)

I added a code comment to clarify this, similar to [here](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/9ae60dc03af6c32ce90734c264461f7ab650b373/dataflow/snippets/src/main/java/com/example/dataflow/BigQueryWrite.java#L56).